doc/source/data/api/dataset_pipeline.rst

```diff
@@ -0,0 +1,177 @@
+.. _dataset-pipeline-api:
+
+DatasetPipeline API
+===================
+
+.. autoclass:: ray.data.dataset_pipeline.DatasetPipeline
+
+**Basic Transformations**
+
+.. autosummary::
+    :nosignatures:
+
+    ray.data.DatasetPipeline.map
+    ray.data.DatasetPipeline.map_batches
+    ray.data.DatasetPipeline.flat_map
+    ray.data.DatasetPipeline.foreach_window
+    ray.data.DatasetPipeline.filter
+    ray.data.DatasetPipeline.add_column
+    ray.data.DatasetPipeline.drop_columns
+
+**Sorting, Shuffling, Repartitioning**
+
+.. autosummary::
+    :nosignatures:
+
+    ray.data.DatasetPipeline.sort_each_window
+    ray.data.DatasetPipeline.random_shuffle_each_window
+    ray.data.DatasetPipeline.randomize_block_order_each_window
+    ray.data.DatasetPipeline.repartition_each_window
+
+**Splitting DatasetPipelines**
+
+.. autosummary::
+    :nosignatures:
+
+    ray.data.DatasetPipeline.split
+    ray.data.DatasetPipeline.split_at_indices
+
+**Creating DatasetPipelines**
```

Comment:
I feel like there's a better name for this section, but I haven't been able to think of one.